chapter8.adoc

```diff
@@ -1,37 +1,124 @@
-Security Model recommendation SR_GEN_007 and SR_GEN_012). When privilege is `M`,
-the `medbgen` gates the `haltreq` from the debug module and if is 0 prevents
-the hart from entering external debug mode. When privilege is less than `M`, the
-OR of the `MSDCFG.sdedbgalw` and `medbgen` gates the `haltreq` from the debug
-module and the hart will enter external debug mode if either is 1.
-
-The configuration for `MSDCFG.sdedbgalw` may be obtained from the manifest/
-configuration of the supervisoer domain and should be managed by the M-mode root
-security manager using secure memory.
-
-When `medbgen` is 1, there are no restrictions. When `medbgen` is 0 but
-`MSDCFG.sdedbgalw` is 1, then the external debug mode may be entered but is
-restricted by `M-mode` software to prevent privileged CSR accesses and memory
-accesses by instructions executed in external debug mode cannot use `M` privilege.
-When `medbgen` is 0 and `MSDCFG.sdedbgalw` is also 0, the M-mode root domain
-security manager must not configure triggers with `action=1`. Triggers for the
-supervisor domain are expected to be controlled by the root domain security
-manager to prevent any leakage of information.
+=== `Smsdetrc`: External Trace allowed control for Supervisor Domain
+
+When M-mode external trace is enabled, all supervisor domains activity may also
+be traced by an external trace tool irrespective of the configuration held in
+`msdcfg.sdetralw`.
+
+When M-mode external trace is disabled, whether execution at privilege modes
+less than `M-mode` may be traced by an external trace tool depends on the
+configuration held in `msdcfg.sdetrcalw`, as described below:
+
+When `msdcfg.sdetrcalw` = 0, external trace is disallowed for the SD. All trace
+output from the trace encoder module are disabled while the SD is active on a
+per-hart basis.
+
+When `msdcfg.sdetrcalw` = 1 then external trace of privilege modes less than
+`M-mode` shall be allowed for the SD on a per hart basis.
+
+To enforce the above controls specified by this extension, the following
+functional and security requirements must be met by the external secure debug
+system cite:[ExtDbgSec] to meet the security objectives of supervisor domain
+isolation:
+
+. The enable control for external trace driven by the external trace module is
+  expected to be established by the platform root-of-trust following RISC-V
+  Security Model recommendations SR_GEN_007 and SR_GEN_012. When the control is
+  in cleared state, the hart should not be able to generate trace.
+. Per the Efficient trace specification cite:[ETrc], the side-band `halted`
+  signal being asserted, stops subsequent tracing from the hart. On this signal
+  being deasserted, the encoder can start tracing again. Implementation of this
+  `halted` side-band signal is required to support external tracing with
+  supervisor domains. The state of the `halted` side-band signal shall be gated
+  by the external tracing enable control for M-mode.
```

Comment:
This detail is not required. Suffices to say trace is disallowed for this section.

Reply:
ack